install_files/ansible-base/roles/common/tasks/harden_dns.yml

```diff
@@ -6,3 +6,22 @@
   tags:
     - dns
     - hardening
+
+- name: Disable systemd-resolved on Focal
+  systemd:
+    name: systemd-resolved
+    state: stopped
+    enabled: no
+  when: ansible_distribution_release == 'focal'
+  tags:
+    - dns
+    - hardening
+
+- name: Create resolv.conf for Focal
+  template:
+    src: dns_base
+    dest: /etc/resolv.conf
+  when: ansible_distribution_release == 'focal'
+  tags:
+    - dns
+    - hardening
```

Comment:
No objections to disabling `systemd-resolved`, sticking with resolvconf which we've been using for a while will be fairly straightforward. Since we just copied the same `dns_base` source a few lines above, let's use that task to write the file. Sounds like on Xenial we want the /etc/resolvconf/resolve.conf.d/ path, whereas on Focal we should write it directly to `/etc/resolv.conf`.

Reply:
Added a commit to consolidate the logic here a bit: under Focal, the old `/etc/resolvconf/resolve.conf.d/` path is no longer written to, and the tests now inspect the correct file based on distro.